iree/turbine/kernel/wave/codegen.py

```diff
@@ -652,9 +652,9 @@ def handle_self_index(emitter: WaveEmitter, node: fx.Node):
         index_type,
         get_constant_attr(cast_py_literal(emitter, stride), index_type))
     stride_vec = vector_d.splat(vector_index_type, stride_cst)
-    scaled = arith_d.MulIOp(step, stride_vec)
+    scaled = arith_d.muli(step, stride_vec)
     offset = vector_d.splat(vector_index_type, start)
-    shifted = arith_d.AddIOp(scaled, offset)
+    shifted = arith_d.addi(scaled, offset)
```

Comment:
if you prefer the pretty form, should you also use 
```
arith_d.index_cast
```
below?

Reply:
makes sense, done!